cmd/osmosisd/cmd/balances_from_state_export.go

```diff
@@ -316,37 +318,18 @@ Example:
 				snapshotAccs[addr] = account
 			}
 
-			/*
-				Structure:
-				for addr, account := range snapshotAccs {
-
-					acc := keeper.GetAccount(ctx, account.Address)
-
-					err := *authtypes.ModuleAccount(acc)
-					//If there IS an error (meaning acc is NOT a module account), continue to next acc (i.e. don't remove)
-					if err != nil {
-						continue
-					}
-
-					//Repeat same casting/error check but w/o * in front of the cast, just to be comprehensive
-					err := authtypes.ModuleAccount(acc)
-					//If there IS an error (meaning acc is NOT a module account), continue to next acc (i.e. don't remove)
-					if err != nil {
-						continue
-					}
-
-					//Else, remove account from list - how do we remove an item from a map in go?
-					//ASSUMING THAT "addr" IS THE KEY
-					delete(snapshotAccs, addr)
-				}
-				//Make sure the length etc. is updated so the resulting json file generated next in the func doesn't error/generate garbage
-			*/
+			app := app.Setup(false)
+			ctx := app.BaseApp.NewContext(false, tmproto.Header{Height: 1, Time: time.Now().UTC()})
 
 			for addr, account := range snapshotAccs {
 
-				acc := keeper.accountKeeper.GetAccount(ctx, account.Address)
+				accAddr, err := sdk.AccAddressFromBech32(account.Address)
+				if err != nil {
+					return err
+				}
+				acc := app.AccountKeeper.GetAccount(ctx, accAddr)
 
-				err := *authtypes.ModuleAccount(acc)
+				err = *authtypes.ModuleAccount(acc)
```

Comment:
@ValarDragon @mattverse This line isn't compiling and gives the error: "cannot convert acc (variable of type "github.com/cosmos/cosmos-sdk/x/auth/types".AccountI) to "github.com/cosmos/cosmos-sdk/x/auth/types".ModuleAccount"

The original approach [we had discussed in the issue](https://github.com/osmosis-labs/osmosis/issues/479#issuecomment-1038720585) was to attempt to typecast each account as a ModuleAccount and check if it works/throws an error. Does this compiler error here mean this approach isn't valid, or is there a way around this?